### PR TITLE
app-office/drawio-desktop-bin: new package, add 22.1.16

### PR DESCRIPTION
**Pull request by MocioF**

### CODE REVIEW

File: app-office/drawio-desktop/drawio-desktop-22.1.2.ebuild
Comment (Flowdalic) [inline app-office/drawio-desktop/drawio-desktop-22.1.2.ebuild:49]: missing `|| die`

besides: why is this needed? I'd assume the permissions are already correct?

File: app-office/drawio-desktop/drawio-desktop-22.1.2.ebuild
Comment (Flowdalic) [inline app-office/drawio-desktop/drawio-desktop-22.1.2.ebuild:55]: The `drawio-desktop-edit.sh` script is small enough that it can be created by a bash here doc in `src_install()`. This avoids having to copy'ing it over and sed'ing it.

**Review by Flowdalic — CHANGES_REQUESTED** [review]
Minor remarks, please squash commits when ready for next review.

File: app-office/drawio-desktop/drawio-desktop-22.1.2.ebuild
Comment (MocioF) [reply app-office/drawio-desktop/drawio-desktop-22.1.2.ebuild:49]: Using previous versions of node_modules' tarball it was necessary. Now it seems not to be needed.
I will remove the line, and I will monitor this issue on future releases.

**Comment by Flowdalic** [comment]
Could you cherry-pick https://github.com/Flowdalic/gentoo/commit/345d95d1166e0c61228a6456090bd79f35e2c9b3 on top of this PR.

I am going to ask my fellow proxy maintainers about this. It is a huge electron app and basically a binary package (as far as I understand, but on the other hand, it seems quite a popular app.

File: app-office/drawio-desktop/drawio-desktop-22.1.2.ebuild
Comment (eli-schwartz) [inline app-office/drawio-desktop/drawio-desktop-22.1.2.ebuild:19]: I can't check right now, but from a glance this appears to be *all* the node_modules, pregenerated.

Including electron. Electron is a compiled binary so that would mean this distfile is arch specific. It would need to conditionalize that specifically, as well as avoid keywording arches that don't have a pregenerated distfile.

File: app-office/drawio-desktop/drawio-desktop-22.1.2.ebuild
Comment (parona-source) [inline app-office/drawio-desktop/drawio-desktop-22.1.2.ebuild:18]:
```suggestion
	https://github.com/jgraph/drawio/archive/refs/tags/v${DRAWIO_V}.tar.gz -> ${PN}-${DRAWIO_V}.tar.gz
```

v22.1.4.tar.gz is very generic

File: app-office/drawio-desktop/drawio-desktop-22.1.2.ebuild
Comment (parona-source) [inline app-office/drawio-desktop/drawio-desktop-22.1.2.ebuild:55]: https://devmanual.gentoo.org/general-concepts/filesystem/index.html

/opt is mean for packages that cant installed normally.  

>  The /opt top-level should only be used for applications that do not conform to the standard filesystem layout. This particularly includes prebuilt software packages that expect being installed in a single directory. 

> The /usr/share directory is for architecture independent application data which is not modified at runtime.

**Comment by parona-source** [comment]
Locales may need chromium-2 handling.

https://devmanual.gentoo.org/eclass-reference/chromium-2.eclass/index.html

Also would it be better to package the binary (deb/rpm) instead similar to discord and the rest.

**Comment by Flowdalic** [comment]
> Also would it be better to package the binary (deb/rpm) instead similar to discord and the rest.

If that is feasible then we should go this way.

**Comment by MocioF** [comment]
This is a bin version of the ebuild, that uses .deb packages.
I have 3 questions:
1) in guru there is a drawio-desktop-bin ebuild using AppImage; is this a problem?
2) in this way we cannot update to fresh jgraph/drawio releases but we have to wait for jgraph/drawio-desktop to be released; is it ok?
3) the app cannot run if a $DISPLAY is not found; should this Rdepend on x11-base/xorg-server or something else?

**Review by Flowdalic — CHANGES_REQUESTED** [review]
> 1. in guru there is a drawio-desktop-bin ebuild using AppImage; is this a problem?
> 2. in this way we cannot update to fresh jgraph/drawio releases but we have to wait for jgraph/drawio-desktop to be released; is it ok?
> 3. the app cannot run if a $DISPLAY is not found; should this Rdepend on x11-base/xorg-server or something else?

Re 1.: I think we will just remove the drawio version from ::guru once this lands in ::gentoo. Feel free to reach out to the GURU maintainer about collaborating on drawio's maintenance in ::gentoo.

Re 2. "In this way" =? AppImage. If so, then the question is probably answered by not using the AppImage approach in ::gentoo.

Re 3.: Not a wayland expert, but I would assume wayland has a X11 compatibility layer which can launch non-wayland apps? I don't think we have to restrict drawio to X11.

File: app-office/drawio-desktop-bin/drawio-desktop-bin-22.1.2.ebuild
Comment (MocioF) [reply app-office/drawio-desktop-bin/drawio-desktop-bin-22.1.2.ebuild:37]: I checked all licenses for package in node_modules. It seems we don't need these restrictions. I will remove them, and update licenses.

File: app-office/drawio-desktop-bin/drawio-desktop-bin-22.1.2.ebuild
Comment (MocioF) [reply app-office/drawio-desktop-bin/drawio-desktop-bin-22.1.2.ebuild:29]: No, I don't have the hardware. I would keep it in the package, just waiting for someone who can test it.

Comment (parona-source) [reply app-office/drawio-desktop-bin/drawio-desktop-bin-22.1.2.ebuild:29]: The policy is to request keywording on bugzilla afterwards if you cant test it yourself. 

(This doesn't mean requesting keywording on every arch under the sun of course, too much work so someone actually using it would be good.)

Comment (MocioF) [reply app-office/drawio-desktop-bin/drawio-desktop-bin-22.1.2.ebuild:27]: Can I leave the arch-conditional distfile in SRC_URI if I remove the ~arm64 keyword?

Comment (parona-source) [reply app-office/drawio-desktop-bin/drawio-desktop-bin-22.1.2.ebuild:27]: You can, it'll just be there ready for the keywording then.

File: app-office/drawio-desktop-bin/drawio-desktop-bin-22.1.2.ebuild
Comment (Flowdalic) [inline app-office/drawio-desktop-bin/drawio-desktop-bin-22.1.2.ebuild:31]: What you call BSD-3 here is just BSD in ::gentoo.

File: licenses/BSD-3
Comment (Flowdalic) [inline licenses/BSD-3:1]: This file can be dropped, as it appears identical to `::gentoo/license/BSD`.

File: app-office/drawio-desktop-bin/metadata.xml
Comment (Flowdalic) [inline app-office/drawio-desktop-bin/metadata.xml:10]: Did Vitaly agree to be added?

Since he is not yet a proxied maintainer in Gentoo, I would suggest to drop him for now and add him once he contributed to this package in ::gentoo.

File: app-office/drawio-desktop-bin/drawio-desktop-bin-22.1.2.ebuild
Comment (Flowdalic) [inline app-office/drawio-desktop-bin/drawio-desktop-bin-22.1.2.ebuild:69]:
```suggestion
	pushd "opt/drawio/locales" > /dev/null || die
```

File: app-office/drawio-desktop-bin/drawio-desktop-bin-22.1.2.ebuild
Comment (Flowdalic) [inline app-office/drawio-desktop-bin/drawio-desktop-bin-22.1.2.ebuild:71]:
```suggestion
	popd > /dev/null || die
```

**Review by Flowdalic — CHANGES_REQUESTED** [review]

File: app-office/drawio-desktop-bin/metadata.xml
Comment (MocioF) [reply app-office/drawio-desktop-bin/metadata.xml:10]: I talked with him and he agree to be included as a maintaner

Comment (MocioF) [reply app-office/drawio-desktop-bin/metadata.xml:10]: Ok @Flowdalic, I will remove him from metadata to include him later if he will contribute on this ebuild.

**Pull request closed without merging**
